src/main/ui/app/route/SearchPage.jsx

```diff
@@ -77,13 +73,15 @@ export default class SearchPage extends Component {
     }
 
     render() {
+        let query = this.props.location.query;
+
         if (this.state && this.state.error) {
             return <div>{this.state.error}</div>
         }
         else {
             return (
                     <div>
-                        <SearchBox initialQuery={this.props.location.query.q}/>
+                        <SearchBox initialQuery={`${query.type ? "type:" + query.type : ""} ${query.q ? "query:" + query.q : ""} ${query.tag ? "tag:" + query.tag : ""} ${query.exclude ? "-" + query.exclude : ""} ${query.name ? "name:" + query.name : ""} ${query.status ? "status:" + query.status : ""} ${query.moduleid ? "moduleId:" + query.moduleid : ""} ${query.uid ? "uid:" + query.uid : ""}`}/>
```

Comment:
The fact that you're doing this feels like a code smell.

Maybe this would work better if SearchBox continues to just be mainly a UI element, and you move the parsing into this SearchPage class?